### DISCUSSION TITLE
How can I use the new variables of a form in a flow?

### DISCUSSION THREAD

**Discussion by wsw70** [windmill-labs/windmill#2749] [Questions] [answered]
I started with the basic flow and added two string inputs `input1` and `input2`. I found them in the auto-generated UI.

In that UI I wanted to add a text area that I called `someText`. I expected it to become available in the flow as an extra input but it is nowhere to be found.

What should I do so that the contents of such components (I will also use checkmarks) are available in the flow that the UI is based from?

**Comment by rubenfiszel** [comment]
you would need to add them to the flow inputs in the flow builder (like you did for input1 and input2), and then in the UI builder, connect those flow inputs to your UI inputs (they will now appear in the right builder panel)

**Reply by wsw70** [reply]
Thank you for your answer (super fast answer :))

It may be easier if I show the screenshots:

![image](https://github.com/windmill-labs/windmill/assets/1345886/aab2dbed-053c-48c6-9ad8-f05f69b21157)


![image](https://github.com/windmill-labs/windmill/assets/1345886/06a6a14b-d38c-4f3c-b52f-53af39d7cede)

> in the UI builder, connect those flow inputs to your UI inputs (they will now appear in the right builder panel)

This is the part I do not understand: if I add `someText` in the flow, it will become available in the UI within the autogenerated form. I do not understand how to create a component which content would go into that variable (in the screenshots above I wanted to reuse `input1` which is already defined in the flow, I did not add `someText` to the flow yet)

**Reply by rubenfiszel** [reply, marked as answer]
use eval or the plug to connect it dynamically to that component output on the left panel or clicking on the red plug when in connection mode. Would recommend following the tutorials in "Tutorials" above.

**Reply by wsw70** [reply]
@rubenfiszel sorry, I should have mentioned that I read the (very good) documentation, I assume that you refer to https://www.windmill.dev/docs/apps/app_editor and especially https://www.windmill.dev/docs/apps/connecting_components?

I re-read them and realized that I was missing a few steps (not to mention that I forgot once to deploy and was testing an older version)

1. create an input in the flow (I am not sure if the type matters). Say, `hello`.
2. create a component in the UI, name it to, say, `myInput`
3. select the runnable (the flow above)
4. in the right panel look for the `hello`, click on the plug and select in the right panel the result of `myInput`

I am writing this just in case I forget again.

In retrospect this makes sense (I am just not sure about the initial type of `hello` - since the component also has a type I guess it does not matter), it was just not completely obvious from the documentation.

Anyway - thank you for following up, I managed to have the content of a component, when submitted, being picked up by a flow.
